docs/querying/functions.md

```diff
@@ -193,12 +193,19 @@ Use `histogram_count` in the following way to calculate a rate of observations
 
     histogram_count(rate(http_request_duration_seconds[10m]))
 
-The additional use of `histogram_sum` enables the calculation of the average of
-observed values (in this case corresponding to “average request duration”):
+## `histogram_avg()`
```

Comment:
I think it's worth noting that `histogram_avg(avg_over_time(sum(rate(...)))` means something very different than `avg_over_time(histogram_avg(sum(rate(...)))` because the second query takes an average of averages, so for example: assume we have a histogram with count=sum=1  and another with sum=10, count=100 , then the first query gives ~0,11 and the second 0.55.  This is especially important as we trained people to calculate first then aggregate, see the `rate` function documentation.

Reply:
I see your point about the importance of the function order, however, I’m struggling to create an example with several layers of average functions which would not be very artificial.

Thinking something like this:
```

Note that in some cases aggregation functions should come first to get a meaningful result. For example, to get 
a 2-hour average request duration, the query should look like this:

    histogram_avg(avg_over_time(sum(rate(http_request_duration_seconds[5m]))[2h:])

instead of this query which would result in an average of averages:

    avg_over_time(histogram_avg(sum(rate(http_request_duration_seconds[5m]))[2h:])
```

However, I would rather have a query like this to get the 2h average duration instead:
```
histogram_avg(sum(rate(http_request_duration_seconds[2h])))
```

Reply:
That's fair, hopefully the issue with mixing the order won't come up. Good to go from my point of view.